Content.Shared/BarSign/BarSignSystem.cs

```diff
@@ -0,0 +1,61 @@
+using System.Linq;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+
+namespace Content.Shared.BarSign;
+
+public sealed class BarSignSystem : EntitySystem
+{
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly MetaDataSystem _metaData = default!;
+    [Dependency] private readonly SharedUserInterfaceSystem _userInterface = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<BarSignComponent, MapInitEvent>(OnMapInit);
+        Subs.BuiEvents<BarSignComponent>(BarSignUiKey.Key,
+            subs =>
+        {
+            subs.Event<SetBarSignMessage>(OnSetBarSignMessage);
+        });
+    }
+
+    private void OnMapInit(EntityUid uid, BarSignComponent component, MapInitEvent args)
```

Comment:
what do you have against mapinit that you would subject this handler to the pain of being the only one not a Entity<>